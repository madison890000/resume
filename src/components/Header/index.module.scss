.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
}


.basicInfo {
  text-align: right;
}

.email {
  font-style: italic;
  font-size: var(--base-font-size-middle);
}

.cellphone {
  font-style: italic;
  font-size: var(--base-font-size-middle);
}
