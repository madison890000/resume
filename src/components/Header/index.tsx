import React from 'react';
import styles from './index.module.scss';

interface HeaderProps {
    firstName: string;
    lastName: string;
    email: string;
    cellphone: string;
}

const Header = ({ firstName, lastName, email, cellphone }: HeaderProps) => {
    return (
        <header className={styles.header}>
            <h1>{firstName} {lastName}</h1>
            <div className={styles.basicInfo}>
                <div><span className={styles.email}>{email}</span></div>
                <div><span className={styles.cellphone}>{cellphone}</span></div>
            </div>
        </header>
    );
}

export default Header;
