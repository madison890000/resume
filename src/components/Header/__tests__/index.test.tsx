import React from 'react';
import Header from '../index';
import renderer from 'react-test-renderer';

test('renders Header', () => {
    const HeaderDom = renderer.create(
        <Header
            firstName="test firstName"
            lastName="test lastName"
            email="test@gmail.com"
            cellphone="101101101"
        />
    )
        .toJSON();
    expect(HeaderDom)
        .toMatchSnapshot();
});
