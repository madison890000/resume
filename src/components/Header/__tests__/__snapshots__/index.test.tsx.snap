// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header 1`] = `
<header
  className="header"
>
  <h1>
    test firstName
     
    test lastName
  </h1>
  <div
    className="basicInfo"
  >
    <div>
      <span
        className="email"
      >
        test@gmail.com
      </span>
    </div>
    <div>
      <span
        className="cellphone"
      >
        101101101
      </span>
    </div>
  </div>
</header>
`;
